### DISCUSSION TITLE
File Associations don't work when file only have extension(_no filename_)

### DISCUSSION THREAD

**Discussion by parallels999** [apache/netbeans#7241] [Ideas]
### Apache NetBeans version

Apache NetBeans 21

### What happened

On php laravel there is `.env` file, without name, only extension
https://github.com/laravel/laravel/blob/58baff2c70144c7cd4c04eecb51fee5a24e873a8/composer.json#L42
After adding File Association, works on any `*.env` file, except for the one that has no name, only extension(_main one_)
![image](https://github.com/apache/netbeans/assets/109294935/8e51a401-363d-4c45-878b-850fd37080d5)

### Language / Project Type / NetBeans Component

PHP project

### How to reproduce

Just create any file with new extension but without name, only the extension, add the File Association(_I'm using `text/x-apache-conf`_), then File Association does't work on that file

### Did this work correctly in an earlier version?

No / Don't know

### Operating System

Windows

### JDK

11

### Apache NetBeans packaging

Apache NetBeans provided installer

### Are you willing to submit a pull request?

No

**Comment by matthiasblaesing** [comment]
This is not a bug, but works as intended (lines 232 + 239):

https://github.com/apache/netbeans/blob/905ff7ef8344e372318751f6e93082a35da3b87f/platform/masterfs/src/org/netbeans/modules/masterfs/filebasedfs/utils/FileInfo.java#L229-L241

With my linux hat on, I agree with that assessment. Traditionally files with leading dot are hidden files. So from that perspective `.env` is not a file that has only an extension, but a hidden file, that has no extension.

**Comment by parallels999** [comment]
>So from that perspective .env is not a file that has only an extension, but a hidden file, that has no extension.

Yes, on linux, but windows is another story, right?

**Comment by Chris2011** [comment]
In the world of DOT files inside of the projects nowadays (.eslintrc, .prettierrc, .env, .anything) we should handle them too. It is not just a linux/windows thing anymore I would say. Let this convert to a discussion to talk about the cases.

**Reply by cesarreyes3** [reply]
On DOT files are really hard comment/uncomment code blocks, You have to do it line by line depending on the mimetype

**Reply by fredsal** [reply]
Other editors open DOT files with correct highlighting

**Reply by matthiasblaesing** [reply]
Sure and NetBeans could do so too, if people would implement it. The Apache conf example might be very helpful for this.

1. Define the way your mimetype is detected: https://github.com/apache/netbeans/blob/ecc3f234600201cf03a35df3977b09855b91b445/webcommon/languages.apacheconf/src/org/netbeans/modules/languages/apacheconf/resources/ApacheConfResolver.xml#L29-L32
2. Register it in the IDE: https://github.com/apache/netbeans/blob/ecc3f234600201cf03a35df3977b09855b91b445/webcommon/languages.apacheconf/src/org/netbeans/modules/languages/apacheconf/ApacheConfDataObject.java#L43-L47

Registering by name might be exposed in GUI, but that needs work and someone championing the effort.
